main/test/flix/Test.Exp.Match.Record.flix

```diff
@@ -0,0 +1,188 @@
+    def testMatchRecord18(): Bool = match { a = { a = { a = { } }, b = { } }, b = { } } {
+        case { a = { a = { a = { } | _ } | _ } | _ } => true
+    }
+
+    @test
+    def testMatchRecord19(): Bool = match { a = { a = { a = { }, b = { } }, b = { } }, b = { } } {
+        case { a = { a = { a = { } | _ } | _ } | _ } => true
+    }
+
+    @test
+    def testMatchRecord20(): Bool = match { a = { a = { a = { }, b = { } }, b = { } }, b = { } } {
+        case { a = { a = { b = { } | _ } | _ } | _ } => true
+    }
+
+    @test
+    def testMatchRecord21(): Bool = match { a = { a = { a = { }, b = { } }, b = { } }, b = { } } {
+        case { a = { a = { a = { }, b = { } | _ } | _ } | _ } => true
+    }
+
+    @test
+    def testMatchRecord22(): Bool = match { a = { a = { a = { }, b = { } }, b = { } }, b = { } } {
+        case { a = { a = { a = { }, b = { } | _ }, b = { } | _ } | _ } => true
+    }
+
+    @test
+    def testMatchRecord23(): Bool = match { a = { a = { a = { }, b = { } }, b = { } }, b = { } } {
+        case { a = { a = { a = { }, b = { } | _ }, b = { } | _ }, b = { } | _ } => true
+    }
+
+    @test
+    def testMatchRecord24(): Bool = match { a = { a = { a = { }, b = { } }, b = { } }, b = { x = A.A } } {
+        case { a = { a = { a = { } | _ } | _ }, b = { x = A.A } } => true
+        case { a = { a = { a = { } | _ } | _ }, b = { x = A.B } } => false
+    }
+
+    @test
+    def testMatchRecord25(): Bool = match { a = { a = { a = { }, b = { b = A.B } }, b = { } }, b = { x = A.A } } {
+        case { a = { a = { a = { } | _ } | _ }, b = { x = A.A } } => true
+        case { a = { a = { a = { } | _ } | _ }, b = { x = A.B } } => false
+    }
+
+    @test
+    def testMatchRecord26(): Bool = match { a = { a = { a = { a = A.B }, b = { b = A.B } }, b = { } }, b = { x = A.A } } {
+        case { a = { a = { a = { a = A.A } | _ } | _ }, b = { x = A.A } } => false
+        case { a = { a = { a = { a = A.A } | _ } | _ }, b = { x = A.B } } => false
+        case { a = { a = { a = { a = A.B } | _ } | _ }, b = { x = A.A } } => true
+        case { a = { a = { a = { a = A.B } | _ } | _ }, b = { x = A.B } } => false
+    }
+
+    @test
+    def testMatchRecord27(): Bool = match { a = A.A, b = A.A } {
+        case { b = A.A, a = A.A } => true
+        case { b = A.A, a = A.B } => false
+        case { b = A.B, a = A.A } => false
+        case { b = A.B, a = A.B } => false
+    }
+
+    @test
+    def testMatchRecord28(): Bool = match { a = { a = { }, b = A.B }, b = A.A } {
+        case { a = { a = { } | _ } | _ } => true
+    }
+    @test
+    def testMatchRecord29(): Bool = match { a = true, b = true } {
+        case { a | r } => a and r.b
+    }
+
+    @test
+    def testMatchRecord30(): Bool = match { a = false, a = true } {
+        case { a = x, a = y } => (not x) and y
+    }
+
+    @test
+    def testMatchRecord31(): Bool = match { a = false, a = true, a = 42} {
+        case { a = x, a = y | r } => ((not x) and y) and r.a == 42
```

Comment:
Is it correct that `r` should have access to the rightmost field `a` (i.e., `r` refers to the part that was **not** unpacked in the pattern)?

Reply:
Yes, this looks right. You peel of two a's to reveal the 3rd and last a.